### PR TITLE
First Goodie returns pi

### PR DESCRIPTION
**Pull request by jmvbxx**
**What does your Instant Answer do?**

This is my first Goodie and I wanted it to return 'pi' to user-specified number of decimal places.

**What problem does your Instant Answer solve (Why is it better than organic links)?**

I wrote an Instant Answer for this earlier today.

**What is the data source for your Instant Answer? (Provide a link if possible)**

For now pi comes from Math::Trig which only provides limited number of decimal places.

**Why did you choose this data source?**

For simplicity.

**Are there any other alternative (better) data sources?**

We could calculate pi right in the code.

**What are some example queries that trigger this Instant Answer?**

pi 2 => 3.14
pi 5 => 3.14159

**Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)**

Students

**Is this Instant Answer connected to a DuckDuckHack [Instant Answer idea](https://duck.co/ideas)?**

Yes, I wrote an Instant Answer earlier today.

**Which existing Instant Answers will this one supercede/overlap with?**

N/A

**Are you having any problems? Do you need our help with anything?**

The code is extremely basic.  I have not yet done the pi.t and I have to put checks into pi.pm.

**Where did you hear about DuckDuckHack? (For first time contributors)**

N/A

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

img-20141216-wa0003

## Checklist
Please place an 'X' where appropriate.

```
[] Added metadata and attribution information (https://duck.co/duckduckhack/metadata)
[] Wrote test file and added to t/ directory (https://duck.co/duckduckhack/goodie_tests)
[] Verified that instant answer adheres to design guidelines (https://duck.co/duckduckhack/design_styleguide)
[] Verified that instant answer adheres to code styleguide (https://duck.co/duckduckhack/code_styleguide)
[] Tested cross-browser compatibility

    Please let us know which browsers/devices you've tested on:
    - Windows 8
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 10

    - Windows 7
        [] Google Chrome
        [] Firefox
        [] Opera
        [] IE 8
        [] IE 9
        [] IE 10

    - Windows XP
        [] IE 7
        [] IE 8

    - Mac OSX
        [] Google Chrome
        [] Firefox
        [] Opera
        [] Safari

    - iOS (iPhone)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - iOS (iPad)
        [] Safari Mobile
        [] Google Chrome
        [] Opera

    - Android
        [] Firefox
        [] Native Browser
        [] Google Chrome
        [] Opera

### CODE REVIEW

**Comment by jagtalon** [comment]
@jmvbxx Keep us updated! Are you planning on making it work for an arbitrary number of digits? http://www.wolframalpha.com/input/?i=pi+to+10000+digits. We wouldn't want the computation to take too long (or else the page won't load quickly) so a limit of some sort would be good.

**Comment by jagtalon** [comment]
@jmvbxx Any updates?

File: lib/DDG/Goodie/Pi.pm [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# ABSTRACT: This Goodie returns 'pi' to a user-specified number of decimal places
+
```
Comment (jagtalon) [inline lib/DDG/Goodie/Pi.pm]: You need `package DDG::Goodie::Pi;` too

**Comment by moollaza** [comment]
ping! @jmvbxx are you still interested in working on this?

**Comment by moollaza** [comment]
Closing and marking this one as abandoned :cry:

**Comment by puskin94** [comment]
@moollaza Hei, which may be the limit???

**Comment by moollaza** [comment]
@puskin94 I'm not exactly sure, but we should try and have the response time be under 10ms. In the current calculator good, we run the math in an `eval` which has an `alarm` to prevent the calculation from running for too long.

If you want to assess the time it takes for the code to execute https://metacpan.org/release/Devel-NYTProf would probably be a good starting point

**Comment by moollaza** [comment]
I'd say start at 1000 digits and see where that lands!

**Comment by puskin94** [comment]
@moollaza Ok! what workflow must i follow? should I create a new branch or.... How can I continue this IA in the better way??

**Comment by mintsoft** [comment]
@puskin94 you can branch, pull from jmvbxx's fork into your branch and continue to work on it

**Comment by puskin94** [comment]
@mintsoft @moollaza Hi, I noticed that this code works until the number of decimals is under 15, because if you try it with pi 100 the result is always 3.14159265358979
Any suggestions?

**Comment by puskin94** [comment]
@mintsoft @moollaza Should I hardcode a constant PI with fixed decimals and  cut it if needed?

**Comment by mintsoft** [comment]
@puskin94 that'd definitely work

**Comment by moollaza** [comment]
@jmvbxx @puskin94 congrats! It's live: https://duckduckgo.com/?q=24+digits+of+pi&ia=answer

Thanks again for taking the time to contribute! We really appreciate it. Feel free to stick around and comment/help on other PR's & Issues or even submit more Instant Answers!

**Comment by puskin94** [comment]
@moollaza I'm already finishing other abandoned IAs. But I'm just waiting the new Goodie Template for update the 2048 game and push it in production!

**Pull request closed without merging**